classes/i18n/PKPLocale.inc.php

```diff
@@ -146,8 +146,11 @@ static function initialize($request) {
 		// Starting from PHP 5.3.0 PHP will throw an E_WARNING if the default
 		// time zone is not set and date/time functions are used
 		// http://pl.php.net/manual/en/function.date-default-timezone-set.php
-		$timeZone = self::getTimeZone();
-		date_default_timezone_set($timeZone);
+		date_default_timezone_set(
+			Config::getVar('general', 'time_zone') // Load the time zone from the configuration file
+			?: ini_get('date.timezone') // Fall back to the time zone set in php.ini
+			?: 'UTC' // Fall back on UTC
```

Comment:
I've never seen this syntax before. Is it supposed to be:

```
Config::getVar('general', 'time_zone')
  ? ini_get('date.timezone')
  : 'UTC'
```

Reply:
This is the ["Elvis operator"](https://stackoverflow.com/questions/1993409/operator-the-elvis-operator-in-php) (and best-named operator ever) -- it evaluates to the left expression if left is truthy, otherwise right. So it's good shorthand for fallbacks. As currently coded it falls back to one of 3 options -- your expression tests the first and returns always either the second or third.

Reply:
Ah, so is it like a chainable coalescing operator? So is it similar to this (which I think isn't valid):

```php
$something = $a ?? $b ?? $c ?? $d;
```

Reply:
Yes, it's chainable (and your expression is valid too). I suppose we could use the null coalescing operator here too; one scenario where the null coalescing `??` operator is better is where an `isset` is required. The `??` builds that in.

Reply:
:+1: Good to know. I don't have an opinion on which is best just learning about it.